Source/Evolve/WarpXEvolve.cpp

```diff
@@ -255,6 +256,14 @@ WarpX::Evolve (int numsteps)
 
         mypc->ApplyBoundaryConditions();
 
+        m_particle_buffers->gatherParticles(*mypc, amrex::GetVecOfConstPtrs(m_distance_to_eb));
```

Comment:
Shouldn't this be called before the call to `mypc->ApplyBoundaryConditions()`, since that will by default flag particles outside the domain? This wipes out the particle `id` since it will be set to `-1`.

Reply:
Yes, I had forgot that this not done in `Redistribute` any more.

Reply:
Resolved: `mypc->ApplyBoundaryConditions()` is now called in the next line.
Thanks for catching, Dave! :)